modules/editorial-metadata/editorial-metadata.php

```diff
@@ -286,66 +251,114 @@ public static function get_editorial_metadata_term_by( $field, $value ) {
 			return false;
 		}
 
+		$term = false;
+
 		if ( 'id' === $field ) {
-			$field = 'term_id';
+			$term = get_term( $value, self::METADATA_TAXONOMY );
+		} else {
+			$term = get_term_by( $field, $value, self::METADATA_TAXONOMY );
+		}
+
+		return null !== $term ? $term : false;
+	}
+
+	/**
+	 * Add all the metadata fields to a term
+	 *
+	 * @param WP_Term $term The term to add metadata to
+	 * @return WP_Term $term The term with metadata added
+	 */
+	public static function add_metadata_to_term( WP_Term $term ): WP_Term {
+		if ( ! isset( $term->taxonomy ) || self::METADATA_TAXONOMY !== $term->taxonomy ) {
+			return $term;
 		}
 
-		// ToDo: This is inefficient as we are fetching all the terms, and then finding the one that matches.
-		$terms = self::get_editorial_metadata_terms();
-		$term  = wp_filter_object_list( $terms, [ $field => $value ] );
+		// if metadata is already set, don't overwrite it
+		if ( isset( $term->meta ) && isset( $term->meta[ self::METADATA_TYPE_KEY ] ) && isset( $term->meta[ self::METADATA_POSTMETA_KEY ] ) ) {
+			return $term;
+		}
 
-		$term = array_shift( $term );
+		$term->meta = get_term_meta( $term->term_id );
```

Comment:
I have a couple suggestions here:

- `$term->meta` is set before any of the keys or data types are validated. If the `isset()` conditions below are not satisfied, `$term` is still returned with invalid `$term->meta` set. How about using a placeholder variable like `$term_meta` during evaluation, and setting once validation is complete?

- `get_term_meta()` as used here returns all meta values, which could include values we didn't expect. I think calling `get_term_meta()` with a supplied key for each expected metadata value is more idiomatic. Looking into the source of [`get_term_meta()`](https://developer.wordpress.org/reference/functions/get_term_meta/), which eventually calls into [`get_metadata_raw()`](https://github.com/WordPress/wordpress-develop/blob/6.6.2/src/wp-includes/meta.php#L600-L668), the whole metadata for an object is already cached whether the metadata is queried all at once or separately.

Here's a way of doing the same process more explicitly, and we also can avoid `array_shift`ing results:

```php
$term_meta = [];
$term_meta[ self::METADATA_TYPE_KEY ] = get_term_meta( $term->term_id, self::METADATA_TYPE_KEY, true );
$term_meta[ self::METADATA_POSTMETA_KEY ] = get_term_meta( $term->term_id, self::METADATA_POSTMETA_KEY, true );

if ( '' === $term_meta[ self::METADATA_TYPE_KEY ) || '' === $term_meta[ self::METADATA_POSTMETA_KEY ] ) {
	return $term;
}

$term->meta = $term_meta;
```

Reply:
Good catch on not setting `single` to true, that is what caused the array to be returned. 

I like the suggestion you've made, especially about just setting the relevant keys that we are looking for. I had originally done this so we didn't need to keep expanding the list of keys to add but really given how specific our validation check is that's moot right now.

Reply:
Pushed this up